pkg/specs/interface.go

```diff
@@ -210,49 +207,48 @@ func (r *Resolver) ResolveRelease(ctx context.Context, upstream string) (*api.Re
 		return r.resolveInlineShipYAMLRelease(
 			ctx,
 			upstream,
-			localPath,
-			applicationType,
+			app,
 		)
 
 	}
 
-	return nil, errors.Errorf("unknown application type %q for upstream %q", applicationType, upstream)
+	return nil, errors.Errorf("unknown application type %q for upstream %q", app.GetType(), upstream)
 }
 
 // read the content sha without writing anything to state
 func (r *Resolver) ReadContentSHAForWatch(ctx context.Context, upstream string) (string, error) {
 
 	debug := level.Debug(log.With(r.Logger, "method", "ReadContentSHAForWatch"))
 	debug.Log("event", "fetch latest chart")
-	appType, localPath, err := r.appTypeInspector.DetermineApplicationType(ctx, upstream)
+	app, err := r.appTypeInspector.DetermineApplicationType(ctx, upstream)
 	if err != nil {
 		return "", errors.Wrapf(err, "resolve app type for %s", upstream)
 	}
-	debug.Log("event", "apptype.inspect", "type", appType, "localPath", localPath)
+	debug.Log("event", "apptype.inspect", "type", app.GetType(), "localPath", app.GetLocalPath())
 
 	defer func() {
-		if err := r.FS.RemoveAll(localPath); err != nil {
+		if err := app.Remove(r.FS); err != nil {
```

Comment:
what is the reasoning behind passing in the FS here? I would have expected to just see `localPath` replaced with `app.GetLocalPath()` everywhere (which is a really solid refactor IMO). what does this get us over `r.FS.RemoveAll(app.GetLocalPath)`

Reply:
(not saying this is a blocker, just want to understand the motivations, and what problems you may have hit that I'm not thinking of)

Reply:
@dexhorthy `localPath` is the location of the chart within the cloned repo, but we need to remove the entire temp folder, which is some arbitrary number of directories up in the file tree.

Reply:
If you look in that `Remove` function, it doesn't remove localPath but a different path.

Reply:
gotchaaa okay makes sense. Thanks for clarifying